src/Materials/customShaderCode.ts

```diff
@@ -0,0 +1,295 @@
+import { Effect } from "./effect";
+import { Mesh } from "../Meshes/mesh";
+import { Texture } from "./Textures/texture";
+import { MaterialDefines } from "./materialDefines";
+
+/**
+ * Defines a code piece to customize the shader.
+ */
+export class ShaderCustomPiece {
+    /**
+     * The code fragment.
+     */
+     public code: string;
+    /**
+     * The insertion point.
+     */
+    public point: ShaderCodePoint;
+    /**
+     * The priority. Higher numbers are inserted first.
+     */
+    public priority: number;
+    constructor(code: string, point: ShaderCodePoint, priority: number = 0) {

```

Comment:
Can be:

constructor(public code: string, public point: ShaderCodePoint, public priority: number = 0) {